### DISCUSSION TITLE
How do I hide specific cells from specific rows depending on the row's condition on the grid

### DISCUSSION THREAD

**Discussion by dBianchii** [serenity-is/Serenity#6477] [Help]
![image](https://user-images.githubusercontent.com/17372417/182460226-871179cf-a198-4567-82bb-72add1b163dd.png)

I have some actionable icons created using:

`columns.unshift({
                field: "EditRow",
                name: "",
                format: ctx => {
                    return "<a class='inline-action recebimento-action left-arrow-row' title='Receber solicitação'></a>";
                },
                width: 24,
                minWidth: 24,
                maxWidth: 24
            });`.

I would like to hide the icons on specific rows depending on the row's conditions, as well as not make them actionable again. Im struggling to get it to work...

**Comment by minhhungit** [comment]
you should hide it on backend in handler, overwrite OnReturn method of ListHandler

**Reply by petebrieso** [reply]
you can also hide it with css by adding a class to the row based on data, say `hide-arrow` then your css would be something like
`.hide arrow .recebimento-action { visibility:hidden} `

**Reply by dBianchii** [reply]
Which method on the grid.ts file should I add this to?

**Reply by petebrieso** [reply]
```
  protected getItemCssClass(item: xyzRow, index: number): string {
            if (item.myfieldname == "some logic for no arrow") { 
                return "hide-revert-action";
            }
           return '';
    }
```

or you could make a formatter for the column to only produce the arrow when desired.

**Reply by stixoffire** [reply]
In Petes reply 
You need both a css   and the method , css goes in with your site  >content css file, the other code goes in the grid.ts file it should automagically be called I think.

This actually should be how it is called. NOTE super.getItemCssClass(); is first if the method exists as overridable in the base then this will pass the intellisense as good. 
If not then you will need to call the method manually. However I think it is called in base..

  protected getItemCssClass(item: xyzRow, index: number): string {
           super.getItemCssClass();
            if (item.myfieldname == "some logic for no arrow") { 
                return "hide-revert-action";
            }
           return '';
    }
